fact-bounty-client/src/pages/login/actions/authActions.js

```diff
@@ -26,10 +26,12 @@ export const loginUser = userData => dispatch => {
         payload: decoded
       });
     })
-    .catch(err =>
-      dispatch({
-        type: GET_ERRORS,
-        payload: err.response.data
-      })
-    );
+    .catch(err => {
+      if (err && err.response) {
```

Comment:
hi @Tushar19varshney, I don't think we need the check for this, as `err` and `err.response` will always be true I guess.

Reply:
@kartikpandey2
Have a look at this:
![Screenshot from 2019-03-19 21-58-51](https://user-images.githubusercontent.com/34132255/54623872-5ef05480-4a92-11e9-8dfa-aa42742d4b81.png)

In this error comes as data of undefined that means that response is not defined, so that's why I have put the null checks so that code will not breaks.
I hope you get the point :slightly_smiling_face:

Reply:
@Tushar19varshney  ok, but how shall I reproduce this error and this error is using node server or flask server in backend.?

Reply:
@kartikpandey2 I have mentioned the steps in issue #194 and I am using backend server as we are going to shift our backend to Flask

Reply:
@Tushar19varshney I was able reproduce the error and found the error is thrown because of a change in the server response, earlier jwt token field name was `token` now it is `access_token`. I believe it will be better if we try to replicate the `nodejs server response convention and variables name also in flask server`. As this will prevent frontend from breaking and making extra changes to fix the problems which will be arising due to change in response. 

Also, I recommend it will be better if you can replace 
`if(err && err.response)` with 
```
if(err.response || err.request) {
let payload = err.request || err.response.data;
      if (typeof payload === "string") {
        payload = { fetch: payload };
      }
    "dispatch action"
}
```
This consider your case and It will also fix 1 case which I was not able to think( i.e `When server hangs up and don't reply`). You can also refer [Error Handling in axios](https://github.com/axios/axios#handling-error) 😃

Reply:
Why are you equating `err.request` to the payload because `err.request` is in format like:
![Screenshot from 2019-03-20 01-50-44](https://user-images.githubusercontent.com/34132255/54639170-b9011200-4ab2-11e9-8bd9-69572c8a7a42.png)
and Why we are checking for `err.request` ?

Reply:
As you have also mentioned the check of `err.response`, I think it `if(err&&err.response)` would not break as such :slightly_smiling_face:

Reply:
@Tushar19varshney  we are checking for `err.request` just to handle the error due to no response from the server.
I am setting payload = err.request because error 

- Should be object otherwise PropTypes will throw error 
- And error.request is an instance of XHR request and may help will debugging(Just to be in safe side) 😃

Reply:
I think that setting the whole `request` as a payload is not standard as we are only interested in the error message:sweat_smile: and if we are implementing that if `response` is not coming from the server then the payload is equal to `request` then it should be like this:
```js
let payload

if(err && err.response){
    payload = err.response.data
    if(typeof payload == 'string'){
        payload = { message: payload }
    }
}
else{
     payload = err.request.responseText
     if(typeof payload == 'string'){
         payload = { message: payload }
     }
}
dispatch action
}
```

Reply:
@Tushar19varshney  this too makes sense 😃 Implement this way then.

Reply:
@kartikpandey2 Cool I will soon update my PR :+1: